src/core.c

```diff
@@ -132,7 +132,7 @@ void free_memory(const argon2_context *context, uint8_t *memory,
 #endif
 
 void NOT_OPTIMIZED secure_wipe_memory(void *v, size_t n) {
-#if defined(_MSC_VER) && VC_GE_2005(_MSC_VER)
+#if defined(_MSC_VER) && VC_GE_2005(_MSC_VER) || defined(__MINGW32__)
```

Comment:
I'm not certain why this has been tagged as Visual Studio 2005+, so I've used the `__MINGW__`, rather than changing it to `_WIN32` (NB the 64-bit version of mingw-w64 also defines `__MINGW32__`).